packages/atomic/src/components/recommendations/atomic-recs-interface/atomic-recs-interface.tsx

```diff
@@ -190,17 +191,36 @@ export class AtomicRecsInterface
 
   @Watch('language')
   public updateLanguage() {
+    if (!this.commonInterfaceHelper.engineIsCreated(this.engine)) {
+      return;
+    }
+
+    const {updateSearchConfiguration} = loadSearchConfigurationActions(
+      this.engine
+    );
+    this.engine.dispatch(
+      updateSearchConfiguration({
+        locale: this.language,
+      })
+    );
     this.commonInterfaceHelper.onLanguageChange();
   }
 
   @Watch('analytics')
   public toggleAnalytics() {
+    if (!this.commonInterfaceHelper.engineIsCreated(this.engine)) {
+      return;
+    }
+
     this.commonInterfaceHelper.onAnalyticsChange();
   }
 
   public registerFieldsToInclude() {
     const fields = EcommerceDefaultFieldsToInclude.concat(
-      this.fieldsToInclude.split(',').map((field) => field.trim())
+      this.fieldsToInclude
```

Comment:
Should probably go with the `ArrayProp` utils here, for the fields to include property.

Reply:
Ah true!

Reply:
https://coveord.atlassian.net/browse/KIT-2092